hwilib/devices/bitbox02.py

```diff
@@ -267,31 +294,38 @@ def __init__(self, path: str, password: Optional[str] = None, expert: bool = Fal
                 "The BitBox02 does not accept a passphrase from the host. Please enable the passphrase option and enter the passphrase on the device during unlock."
             )
         super().__init__(path, password=password, expert=expert, chain=chain)
-
-        hid_device = hid.device()
-        hid_device.open_path(path.encode())
-        self.transport = u2fhid.U2FHid(hid_device)
+        simulator = None
+        self.noise_config: BitBoxNoiseConfig = BitBoxNoiseConfig()
+
+        if path != SIMULATOR_PATH:
+            hid_device = hid.device()
+            hid_device.open_path(path.encode())
+            self.transport = u2fhid.U2FHid(hid_device)
+            self.noise_config = CLINoiseConfig()
+        else:
+            simulator = BitBox02Simulator()
+            self.transport = u2fhid.U2FHid(simulator)
         self.device_path = path
 
         # use self.init() to access self.bb02.
         self.bb02: Optional[bitbox02.BitBox02] = None
 
-        self.noise_config: BitBoxNoiseConfig = CLINoiseConfig()
-
     def set_noise_config(self, noise_config: BitBoxNoiseConfig) -> None:
         self.noise_config = noise_config
 
     def init(self, expect_initialized: Optional[bool] = True) -> bitbox02.BitBox02:
         if self.bb02 is not None:
             return self.bb02
 
-        for device_info in devices.get_any_bitbox02s():
+        devs = devices.get_any_bitbox02s()
+        devs.append({'path': SIMULATOR_PATH.encode(), 'serial_number': '', 'product_string': 'bitbox02'})
+        for device_info in devs:
             if device_info["path"].decode() != self.device_path:
                 continue
 
             bb02 = bitbox02.BitBox02(
                 transport=self.transport,
-                device_info=device_info,
+                device_info=None,
```

Comment:
This also changes the behavior for regular BitBox02s, not just the simulator, and requires a certain min. FW version to work. I think that version is quite old so it's okay.

Maybe add a comment that passing `None` here means the device is queried for the relevant info.

Reply:
Okay, adding a comment about this there.